### PR TITLE
Tweak build instructions

### PR DESCRIPTION
**Pull request by RandomInEqualities**
I have these 3 commits that mostly tweaks the windows build instructions to have less text and be more focused. It is easy to build with Visual Studio/Clang/GCC on Windows, but there is a lot of text making it seem complicated. With CMake things are quite simple. 

This mainly does 3 things:

- Rename README.windows to README.mingw.md as it is largely only mingw related.
- Remove dist-win.sh script as I think CMake does this better. And dist-win relies on mingw while there are native compiler support. The releases.yml also uses CMake.
- Moves things around in README.md to make it focused on CMake or configure based on what one want to use.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,6 +138,26 @@ into other projects or build by itself.
...
+export UNIX_STD=2003
+./configure
+make
+```
+
+### mingw-w64 - Windows
+
+LibreSSL builds against relatively recent versions of Mingw-w64, not to be
+confused with the original mingw.org project. Mingw-w64 3.2 or later
+should work. See README.mingw.md for more information.
```
Comment (joshuasing) [inline README.md:159]: I think linking to the `README.mingw.md` file could be helpful and allow people to easily navigate to the file.
```suggestion
should work. See [README.mingw.md](README.mingw.md) for more information.
```

File: README.md [hunk 2]
```diff
@@ -166,6 +138,26 @@ into other projects or build by itself.
...
+```sh
+export UNIX_STD=2003
+./configure
+make
+```
+
+### mingw-w64 - Windows
+
+LibreSSL builds against relatively recent versions of Mingw-w64, not to be
+confused with the original mingw.org project. Mingw-w64 3.2 or later
```
Comment (joshuasing) [inline README.md:158]: "Mingw-w64" appears to often be stylised as MinGW-w64 (see https://www.mingw-w64.org/).
Additionally, I think linking to the MinGW-w64 website may be useful.

```suggestion
LibreSSL builds against relatively recent versions of [MinGW-w64](https://www.mingw-w64.org/), not to be
confused with the original mingw.org project. MinGW-w64 3.2 or later
```

File: README.md [hunk 3]
```diff
@@ -68,35 +68,36 @@ sent to the core team at libressl-security@openbsd.org.
...
+Once you have the source tree prepared, run these commands to build and install:
 
 ```sh
 ./configure   # see ./configure --help for configuration options
 make check    # runs builtin unit tests
 make install  # set DESTDIR= to install to an alternate location
 ```
 
-If you wish to use the CMake build system, use these commands:
+Alternatively it is possible to run `./dist.sh` to prepare a tarball.
```
Comment (joshuasing) [inline README.md:96]:
```suggestion
Alternatively, it is possible to run `./dist.sh` to prepare a tarball.
```

**Comment by RandomInEqualities** [comment]
Thanks for the comments @joshuasing I have updated the review :).

**Review by busterb — APPROVED** [review]
This is fine with me, thanks for the updates.

**Pull request merged**